torch/ao/quantization/fx/utils.py

```diff
@@ -945,3 +953,69 @@ def find_patterns(
         new_input = first_tuple.args[0][last_getitem_index]  # type: ignore[index]
         for user in list(last_getitem.users.keys()):
             user.replace_input_with(last_getitem, new_input)
+
+def _qconfig_satisfies_dtype_config_constraints(
+        qconfig: QConfigAny,
+        dtype_with_constraints: DTypeWithConstraints,
+        is_activation: bool = True) -> bool:
+    """
+    Return whether `qconfig` satisfies the following constraints from the backend,
+    specified through the activation and weight DTypeWithConstraints.
+
+        1. QConfig specified a quantization range that falls within the backend's, if any
+        2. QConfig specified a min scale value that is >= the backend's, if any
+
+    If `is_activation` is True, we check `qconfig.activation`, else we check `qconfig.weight`.
+    If `qconfig` or `dtype_with_constraints.dtype` is None, or the dtypes do not match, return True.
+    """
```

Comment:
can be in a separate PR since this PR is too long, I feel we should cleanup the behavior for None (possibly both in backend_config and qconfig), can we reserve `None` only for use cases that the check doesn't make sense? for example, default `quant_min` and `quant_max` can be calculated based on `dtype` for torch.quint8, torch.quint8 etc. 
`scale_min` can default to `torch.finfo(torch.float32).eps` etc.

I think keeping None here just makes things harder to understand, we'll need to consult documentation to understand what that means exactly, so ideally we should eliminate them.

Reply:
I agree, though I feel that's an existing issue; we have a lot of potential None's in general and it's not documented what they really mean. Let's do it in a separate PR to limit the scope of this one.

Reply:
Filed https://github.com/pytorch/pytorch/issues/85745 to track this